cohortextractor/mssql_utils.py

```diff
@@ -194,3 +193,23 @@ def fetch(self, min_key=None):
                     time.sleep(sleep)
                     sleep *= self.backoff_factor
                     self.cursor = None
+
+
+def database_exceptions():
+    """
+    Returns specific database errors we're interested in catching
+
+    We avoid importing this immediately for similar reasons described above,
+    so we can create a TPPBackend instance without needing a MSSQL
+    database driver installed
+    """
+    try:
+        import pymssql
+
+        return (
+            pymssql._pymssql.OperationalError,
+            pymssql._pymssql.ProgrammingError,
+            pymssql._pymssql.IntegrityError,
+        )
+    except ImportError:
+        return ()
```

Comment:
I guess really it should check the other drivers for database-related errors, but given that we only use pymssql now, it didn't seem worth doing, and my intention was that it woulddri return something that wouldn't every get caught, and db errors raised by other drivers (which presumably wouldn't happen anymore unless someone had swapped in another one for some reason) would get raised normally and exit with the default exit code 1.  I don't know that we want to catch all exceptions here though, that would make it always raise a general database error.  It could return a dummy exception class instead?